jwst/photom/photom.py

```diff
@@ -700,7 +701,13 @@ def photom_io(self, tabdata, order=None):
                 slit.var_rnoise *= conversion**2
             if slit.var_flat is not None and np.size(slit.var_flat) > 0:
                 slit.var_flat *= conversion**2
-            slit.dq[no_cal] = np.bitwise_or(slit.dq[no_cal], dqflags.pixel['DO_NOT_USE'])
+            if no_cal is not None:
+                    if len(slit.dq.shape) == 3:
+                        slit.dq[:,no_cal] = np.bitwise_or(slit.dq[:,no_cal],
```

Comment:
I would suggest using this syntax, which will work for `dq.shape` of 2 or higher (assuming `no_cal` is 2-D):

    slit.dq[..., no_cal] = np.bitwise_or(slit.dq[..., no_cal], dqflags.pixel['DO_NOT_USE'])

Reply:
Thanks for the tip. Updated in the last commit.